### PR TITLE
Feature/demoextras drag and drop

### PR DESCRIPTION
**Pull request by hmaciasc**

### CODE REVIEW

File: cp2/ControlPanel/templates/demoExtras.html [hunk 1]
```diff
@@ -20,14 +20,33 @@ <h2>You are not allowed to edit this demo</h2>
         <li>zip -r demoExtras.zip source_dir</li>
     </ul>
     <form id="editDemoExtras" action="{% url 'add_extras' %}" method="post" enctype="multipart/form-data">
-        {% csrf_token %}
-        <input type="hidden" name="demo_id" value="{{ demo_id }}">
-        <label class="file">
+        <div class="extras-input-container">
+            {% csrf_token %}
+            <input type="hidden" name="demo_id" value="{{ demo_id }}">
+            <input type="hidden" name="title" value="{{ title }}">
```
Comment (kidanger) [inline cp2/ControlPanel/templates/demoExtras.html:26]: I don't think title is required here?

File: cp2/ControlPanel/static_cp/js/demoExtras.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            if(droppedFiles)
+            {
+                Array.prototype.forEach.call(droppedFiles, function(file) {
+                    formData.append(input.getAttribute('name'), file)
+                });
+
+                const queryString = window.location.search;
+                const urlParams = new URLSearchParams(queryString);
+                formData.append('demo_id', urlParams.get('demo_id'))
+                formData.append('title', urlParams.get('title'))
```
Comment (kidanger) [inline cp2/ControlPanel/static_cp/js/demoExtras.js:77]: I think think title is required here

**Review by kidanger — CHANGES_REQUESTED** [review]

**Review by kidanger — CHANGES_REQUESTED** [review]
Can you rebase into one or two commits? (in particular we should not unnecessary merge commits)

**Review by mcolom — COMMENTED** [review]
The code is quite clear.

I have just a question before accepting the PR.
In the HTML we have <input type="hidden" name="demo_id" value="{{ demo_id }}", which is posted to {% url 'add_extras' %}.

Is it possible for a user to simply spoof a different value for demo_id and overwrite the extras of a non-owned demo?

**Review by kidanger — APPROVED** [review]

**Review by mcolom — APPROVED** [review]
Answer from Héctor: the backend checks that you're only changing your own demos.

**Review by Jyotsnarajan — APPROVED** [review]

**Pull request merged**